tests/internal/stream_processor.c

```diff
@@ -54,6 +55,19 @@
 
 #define MP_UOK MSGPACK_UNPACK_SUCCESS
 
+static void initialization_crutch()
```

Comment:
Q: Wouldn't be better to call it config_init or similar to express the intention of this method? also, why not having a common function instead? (I see repeated over tests).

Reply:
Maybe, we should create the common function in flb_tests_common.h or some place?

Reply:
It can and should be properly fixed, this is a crutch, initializing and destroying a global object just for the lateral TLS initialization it does is not right. This function could be defined as a static inline in the shared header but I was half way when thought that it would be nicer not to repeat it everywhere (honestly I wasn't expecting it to be so pervasive) so I leaned on the side of not having to do it all over again for a nit.

As for the naming, I would like to make it even more obvious that this is not the way to do it but that's close to the limit of what's acceptable so...

Reply:
> Maybe, we should create the common function in flb_tests_common.h or some place?

Yeah, that's what I thought a while after I mechanically did it, once again, I didn't expect so many of these to be affected. 
We could export the diffs, curate them and make a new branch that incorporates everything once I'm done with the harder issues (probably later today or on monday).

Reply:
Worst case scenario, once all tests are passing we're left with the simple task of making it pretty which I will probably end up doing to appease my neurosis...